pkg/kots/lint.go

```diff
@@ -344,6 +361,103 @@ func lintWithKubevalSchema(renderedFiles SpecFiles, originalFiles SpecFiles, sch
 	return lintExpressions, nil
 }
 
+func checkIfVersionExists(version string) (bool, error) {
+	url := "http://api.github.com/repos/replicatedhq/kots/commits/%s"
+	token := os.Getenv("GITHUB_API_TOKEN")
+	var bearer = "Bearer " + token
+
+	if !strings.HasPrefix(version, "v") {
+		version = "v" + version
+	}
+
+	rwMutex.RLock()
+	verIsCached := kotsVersions[version]
+	rwMutex.RUnlock()
+
+	if !verIsCached {
+		req, err := http.NewRequest("GET", fmt.Sprintf(url, version), nil)
+		if err != nil {
+			return false, errors.Wrap(err, "failed to create new request")
+		}
+		req.Header.Set("Authorization", bearer)
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if resp.StatusCode != 200 {
+			return false, nil
+		} else if resp.StatusCode == 200 {
+			rwMutex.Lock()
+			kotsVersions[version] = true
+			rwMutex.Unlock()
+		}
+	}
+
+	return true, nil
+}
+
+func lintTargetMinKotsVersions(specFiles SpecFiles) ([]LintExpression, error) {
+	lintExpressions := []LintExpression{}
+	// separate multi docs because the manifest can be a part of a multi doc yaml file
+	separatedSpecFiles, err := specFiles.separate()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to separate multi docs")
+	}
+
+	for _, spec := range separatedSpecFiles {
+		var tv, mv string
+		var tvExists, mvExists bool
+		doc := map[string]interface{}{}
+		if err := goyaml.Unmarshal([]byte(spec.Content), &doc); err != nil {
+			return nil, errors.Wrap(err, "failed to unmarshal spec content")
+		}
+		if doc["apiVersion"] == "kots.io/v1beta1" && doc["kind"] == "Application" {
+			if spec, ok := doc["spec"].(map[interface{}]interface{}); ok {
+				tv, tvExists = spec["targetKotsVersion"].(string)
+				mv, mvExists = spec["minKotsVersion"].(string)
+			}
+		}
+
+		// if no min nor target kots version exists, continue to next file
+		if !mvExists && !tvExists {
+			continue
+		}
+
+		targetVersionlintExpression := LintExpression{
```

Comment:
any reason this isn't just defined on line 437 where it's appended?

Reply:
moved def